src/scipp/core/structured.py

```diff
@@ -1,55 +1,67 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 # @author Simon Heybrock
-from .._scipp.core import DType, _element_keys, _get_elements, _set_elements
+from collections.abc import Iterator
+from typing import Any
 
+from .._scipp.core import _element_keys, _get_elements, _set_elements
+from .cpp_classes import DType, Variable
 
-def _prop(key):
-    def getter(fields):
-        return _get_elements(fields._var, key)
 
-    def setter(fields, x):
-        return _set_elements(fields._var, key, x)
+def _prop(key: str) -> property:
+    def getter(self: Any) -> Variable:
+        return _get_elements(self._var, key)  # type: ignore[no-any-return]
+
+    def setter(self: Any, x: Variable) -> None:
+        _set_elements(self._var, key, x)
 
     return property(getter, setter)
 
 
-def is_structured(obj):
+def is_structured(obj: Variable) -> bool:
+    """Check whether a variable has a structured dtype."""
     if obj.bins is not None:
         return is_structured(obj.bins.constituents['data'])
     return obj.dtype in [DType.vector3, DType.linear_transform3]
 
 
-def _fields(obj):
-    class Fields:
-        def __init__(self, keys, var):
-            self._var = var
-            self._keys = keys
+class Fields:
+    """Accessor for fields of a structured variable."""
+
+    def __init__(self, keys: list[str], var: Variable) -> None:
+        self._var = var
+        self._keys = keys
```

Comment:
Does the `_keys` need to have the reference to the original `keys`?

```suggestion
    def __init__(self, keys: Sequence[str], var: Variable) -> None:
        self._var = var
        self._keys = tuple(keys)
```

Reply:
It doesn't. This was to avoid making a copy because that is not needed here. This class should never be instantiated by anyone outside of this module.